trainer.py

```diff
@@ -122,17 +123,24 @@ def _prune(self, model):
     def prune_trained_model(model, threshold):
         if math.isnan(threshold):
             return
+        v = Verbose()
         # Layer 0 has no input edges, start from layer 1
         pruned_edges = 0
         for i in range(1, len(model.layers)):
             weights = model.layers[i].get_weights()
             input_weights = weights[0]  # weights[1] is the list of node biases
+            weighted_threshold = threshold * np.linalg.norm(input_weights)
             # The incoming edge weights of node N is incoming_edge_weights[N]
-            new_weights = np.where((input_weights < threshold) & (input_weights > -threshold), 0, input_weights)
-            pruned_edges += new_weights.size - np.count_nonzero(new_weights)
+            new_weights = np.where(np.absolute(input_weights) < weighted_threshold, 0, input_weights)
             model.layers[i].set_weights([np.array(new_weights), weights[1]])
-        v = Verbose()
-        v.logger.debug("Pruned {} edges (with threshold {})".format(pruned_edges, threshold))
+            pruned_this_time = new_weights.size - np.count_nonzero(new_weights)
+            pruned_edges += pruned_this_time
+            percent_this_layer = (pruned_this_time * 100) // new_weights.size
```

Comment:
don't you need to divide by the number of edges (new_weights.size) here or something?

Reply:
Yes, it's there @galshachaf

Reply:
oh sorry, thought this was a comment.

Reply:
#חחחח